Engine/DataFeeds/FileSystemDataFeed.cs

```diff
@@ -67,14 +67,16 @@ public IEnumerable<Subscription> Subscriptions
         /// <summary>
         /// Initializes the data feed for the specified job and algorithm
         /// </summary>
-        public void Initialize(IAlgorithm algorithm, AlgorithmNodePacket job, IResultHandler resultHandler, IMapFileProvider mapFileProvider, IFactorFileProvider factorFileProvider, IDataProvider dataProvider)
+        public void Initialize(IAlgorithm algorithm, AlgorithmNodePacket job, IResultHandler resultHandler,
+                               IMapFileProvider mapFileProvider, IFactorFileProvider factorFileProvider,
+                               IDataProvider dataProvider, SubscriptionCollection subscriptionCollection)
```

Comment:
Perhaps instead of sending in the `SubscriptionCollection` we should be sending in the `DataManager` -- or perhaps `IDataManager` ?

_Eventually_ the storage should be hidden (encapsulation) and the required behaviors exposed.

Reply:
good point! I've updated it to receive the new `IDataFeedSubscriptionManager`